File: cpp_merge/packages/integration/SWEagent/src/environment/hooks/status.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/SWEagent/src/environment/hooks/status.h"

SetStatusEnvironmentHook::SetStatusEnvironmentHook(string id, StatusCallback callable) : EnvHook() {
    this->id = id;
    this->callable = callable;
}

void SetStatusEnvironmentHook::update(string message)
{
    this->callable(this->id, message);
}

void SetStatusEnvironmentHook::onStartDeployment()
{
    this->update(std:("Starting deployment"));
}

void SetStatusEnvironmentHook::onInstallEnvStarted()
{
    this->update(std:("Installing environment"));
}

void SetStatusEnvironmentHook::onEnvironmentStartup()
{
    this->update(std:("Starting environment"));
}

void SetStatusEnvironmentHook::onClose()
{
    this->update(std:("Closing environment"));
}

File: SWEagent/src/environment/hooks/status.ts
/**
 * Status environment hook
 * Converted from sweagent/environment/hooks/status.py
 */

import { EnvHook } from './abstract';
import { Repo, RepoConfig } from '../repo';

/**
 * Status update callback type
 */
export type StatusCallback = (id: string, message: string) => void;

/**
 * Environment hook that updates status messages
 */
export class SetStatusEnvironmentHook extends EnvHook {
  private callable: StatusCallback;
  private id: string;

  constructor(id: string, callable: StatusCallback) {
    super();
    this.id = id;
    this.callable = callable;
  }

  private update(message: string): void {
    this.callable(this.id, message);
  }

  onCopyRepoStarted(repo: Repo | RepoConfig): void {
    const repoName = 'repoName' in repo ? repo.repoName : 'unknown';
    this.update(`Copying repo ${repoName}`);
  }

  onStartDeployment(): void {
    this.update('Starting deployment');
  }

  onInstallEnvStarted(): void {
    this.update('Installing environment');
  }

  onEnvironmentStartup(): void {
    this.update('Starting environment');
  }

  onClose(): void {
    this.update('Closing environment');
  }
}
